src/main/java/mods/railcraft/common/util/collections/ItemKey.java

```diff
@@ -12,22 +12,23 @@
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraftforge.oredict.OreDictionary;
+import net.minecraftforge.registries.IRegistryDelegate;
 
 /**
  * @author CovertJaguar <http://www.railcraft.info>
  */
 public final class ItemKey {
 
-    public final Item item;
+    public final IRegistryDelegate<Item> item;
```

Comment:
What is this and why is it necessary?

Reply:
This is to handle cases of registry substitution e.g. when a mod replaces a vanilla item. Forge has recommended doing so because mods may replace vanilla objects just like how we replace vanilla carts. This is also done in fluid container handling in the `fluid` package as well.